celery_singleton/singleton.py

```diff
@@ -126,6 +136,14 @@ def lock_and_run(self, lock, *args, task_id=None, **kwargs):
                 # Clear the lock if apply_async fails
                 self.unlock(lock)
                 raise
+    
+    def challenge_lock(self, lock, args, kwargs):
+        current_task_lock = self.generate_lock(self.name, args, kwargs)
+        active_tasks = inspect_celery.list_celery_tasks_with_name(self.celery_app, self.name)
+        active_tasks_lock = [self.generate_lock(task.name, task.args, task.kwargs) for task in active_tasks]
```

Comment:
I'm concerned about performance of this. Generating the hash for every task (say you have thousands) every time you enqueue a new task could be problematic.

Why not get the value of current task lock from redis (`singleton_backend.get(lock)`) this returns the corresponding task ID. You could then check if that task ID is in active tasks, right?

Reply:
Thank you for pointing this out. ID comparison seems a better fit for this function, for both performances and simplicity (I will provide a new implementation)